compiler/src/iree/compiler/DispatchCreation/test/split_argmax_reduction.mlir

```diff
@@ -0,0 +1,71 @@
+// RUN: iree-opt --pass-pipeline='builtin.module(util.func(iree-dispatch-creation-split-reduction-ops,cse, canonicalize))' %s | FileCheck %s
+
+util.func public @argmax(%arg0: tensor<?x131072xbf16>, %arg1: index) -> tensor<?xi64> {
+  %cst = arith.constant 0xFF80 : bf16
+  %c0_i64 = arith.constant 0 : i64
+  %0 = tensor.empty(%arg1) : tensor<?xbf16>
+  %1 = linalg.fill ins(%cst : bf16) outs(%0 : tensor<?xbf16>) -> tensor<?xbf16>
+  %2 = tensor.empty(%arg1) : tensor<?xi64>
+  %3 = linalg.fill ins(%c0_i64 : i64) outs(%2 : tensor<?xi64>) -> tensor<?xi64>
+  %4:2 = linalg.generic {
+      indexing_maps = [affine_map<(d0, d1) -> (d0, d1)>,
+                       affine_map<(d0, d1) -> (d0)>,
+                       affine_map<(d0, d1) -> (d0)>],
+      iterator_types = ["parallel", "reduction"]}
+      ins(%arg0 : tensor<?x131072xbf16>) outs(%1, %3 : tensor<?xbf16>, tensor<?xi64>) {
+  ^bb0(%in: bf16, %out: bf16, %out_0: i64):
+    %5 = linalg.index 1 : index
+    %6 = arith.index_cast %5 : index to i64
+    %7 = arith.maximumf %in, %out : bf16
+    %8 = arith.cmpf ogt, %in, %out : bf16
+    %9 = arith.select %8, %6, %out_0 : i64
+    linalg.yield %7, %9 : bf16, i64
+  } -> (tensor<?xbf16>, tensor<?xi64>)
+  util.return %4#1 : tensor<?xi64>
+}
+
+// CHECK-LABEL:   util.func public @argmax
+
+// Check identity value preparation
+// CHECK: %[[CST:.+]] = arith.constant 0xFF80 : bf16
+// CHECK: %[[ZERO:.+]] = arith.constant 0 : i64
+// CHECK: %[[FINALVAL_EMPTY:.+]] = tensor.empty(%[[ARG1:.+]]) : tensor<?xbf16>
+// CHECK: %[[FINALVAL:.+]] = linalg.fill ins(%[[CST]] : bf16) outs(%[[FINALVAL_EMPTY]] : tensor<?xbf16>) -> tensor<?xbf16>
+// CHECK: %[[FINALIDX_EMPTY:.+]] = tensor.empty(%[[ARG1]]) : tensor<?xi64>
+// CHECK: %[[FINALIDX:.+]] = linalg.fill ins(%[[ZERO]] : i64) outs(%[[FINALIDX_EMPTY]] : tensor<?xi64>) -> tensor<?xi64>
+
+// Check partial reduction
+// CHECK: %[[EXPAND:.+]] = tensor.expand_shape %arg0 {{\[}}[0], [1, 2]] output_shape [%{{.+}}, 1024, 128] : tensor<?x131072xbf16> into tensor<?x1024x128xbf16>
+// CHECK: %[[INITVAL:.+]] = tensor.empty(%{{.+}}) : tensor<?x1024xbf16>
+// CHECK: %[[FILLVAL:.+]] = linalg.fill ins(%{{.+}} : bf16) outs(%[[INITVAL]] : tensor<?x1024xbf16>) -> tensor<?x1024xbf16>
+// CHECK: %[[INITIDX:.+]] = tensor.empty(%{{.+}}) : tensor<?x1024xi64>
+// CHECK: %[[FILLIDX:.+]] = linalg.fill ins(%{{.+}} : i64) outs(%[[INITIDX]] : tensor<?x1024xi64>) -> tensor<?x1024xi64>
+
+// CHECK: %[[PARTIAL:.+]]:2 = linalg.generic
+// CHECK-SAME: iterator_types = ["parallel", "parallel", "reduction"]
```

Comment:
I missed a comment I had in my mind, we may want to check indexing maps as well.